packages/@aws-cdk/aws-codedeploy/lib/base-deployment-config.ts

```diff
@@ -0,0 +1,159 @@
+  /**
+   * The deployment will target EC2 instances or on-premise servers
+   */
+  SERVER = 'Server',
+
+  /**
+   * The deployment will target a Lambda function
+   */
+  LAMBDA = 'Lambda',
+
+  /**
+   * The deployment will target an ECS server
+   */
+  ECS = 'ECS'
+}
+
+/**
+ * Complete base deployment config properties that are required to be supplied by the implementation
+ * of the BaseDeploymentConfig class.
+ */
+export interface BaseDeploymentConfigProps extends BaseDeploymentConfigOptions {
+  /**
+   * The destination compute platform for the deployment.
+   *
+   * @default ComputePlatform.Server
+   */
+  readonly computePlatform?: ComputePlatform;
+
+  /**
+   * The configuration that specifies how traffic is shifted during a deployment.
+   * Only applicable to ECS and Lambda deployments, and must not be specified for Server deployments.
+   * @default None
+   */
+  readonly trafficRouting?: TrafficRouting;
+}
+
+/**
+ * The base class for ServerDeploymentConfig, EcsDeploymentConfig,
+ * and LambdaDeploymentConfig deployment configurations.
+ *
+ * @resource AWS::CodeDeploy::DeploymentConfig
+ */
+export abstract class BaseDeploymentConfig extends Resource implements IBaseDeploymentConfig {
+  /**
+   * Import a custom Deployment Configuration for a Deployment Group defined outside the CDK.
+   *
+   * @param scope the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
+   * @param deploymentConfigName the name of the referenced custom Deployment Configuration
+   * @returns a Construct representing a reference to an existing custom Deployment Configuration
+   */
+  protected static fromDeploymentConfigName(scope: Construct, id: string, deploymentConfigName: string): IBaseDeploymentConfig {
+    ignore(id);
+    const arn = Stack.of(scope).formatArn({
+      service: 'codedeploy',
+      resource: 'deploymentconfig',
+      resourceName: deploymentConfigName,
+      arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+    });
+    return {
+      deploymentConfigName: deploymentConfigName,
+      deploymentConfigArn: arn,
+    };
+  }
+
+  /**
+   * This method should be used only for static references to predefined deployment configurations,
+   * like EcsDeploymentConfig.ALL_AT_ONCE
+   * @param name the name of the referenced custom Deployment Configuration
+   * @returns a reference to an existing custom Deployment Configuration
+   */
+  protected static deploymentConfig(name: string): IBaseDeploymentConfig {
+    return {
+      deploymentConfigName: name,
+      deploymentConfigArn: arnForDeploymentConfig(name),
+    };
+  }
+
+  /**
+   * The name of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props?: BaseDeploymentConfigProps, additionalProps?: any) {
```

Comment:
Can the props from `additionalProps` be added to `props`? I think that is just the
`minimumHealthyHosts` property from the server deployment config.

Reply:
Done!